### PR TITLE
Grouflon master fixes

### PR DESCRIPTION
**Pull request by berdal84**
Few fixes regarding to our recent discussions.
What I have fixed:

- enable MIRROR_API only for WIN32
- MetaData: init member pointers + copy constructor
- fix template declaration order (maybe messy since I have no knowledge of your code)
- lazy init for g_typeSet (I don't know if what I did is correct).
- some small fixes

What I haven't fixed:
- a derived can't be MIRROR_CLASS_DEFINED before its parent. Type have to be stored as TypeId (VS pointer).

### CODE REVIEW

**Comment by berdal84** [comment]
I think those fix aren't sufficient to be able to run any program using mirror as a static library and reflecting primitive types.
User needs to have the primitive `TypeDesc` to be able to execute any cpp file using a `MIRROR_CLASS_DEFINITION`.
I tried to reflect a struct with a single float member, it failed.

Here, my struct portion:
image

The runtime error:
image


Maybe, could be interesting to add a `MIRROR_INIT` macro or a `mirror::Init()` function, to be inserted before anything in `main.cpp`. For me, line 20:

image

**Comment by berdal84** [comment]
Me again, I moved the primitive types definition to main.cpp, it works now:

image

with:

image




I think I can make sense to have:

1. Init mirror
2. Init user classes
3. main()

File: mirror_base.cpp [hunk 1]
```diff
@@ -73,18 +73,21 @@ namespace mirror
...
 	TypeDesc* FindTypeByName(const char* _name)
 	{
-		return ::mirror::g_typeSet.findTypeByName(_name);
+		return GetTypeSet()->findTypeByName(_name);
 	}
 
 	MetaData::MetaData(const char* _name, const char* _data)
+    :
+    m_name(nullptr),
+    m_data(nullptr)
```
Comment (Grouflon) [inline mirror_base.cpp:90]: What is the use of adding that ? the variables are initialized two lines after

Comment (berdal84) [reply mirror_base.cpp:90]: Oh yeah... I think I did that without thinking.

**Pull request closed without merging**